plugin/tokudb-backup-plugin/tokudb_backup.cc

```diff
@@ -421,21 +436,43 @@ class source_dirs {
         return buf;
     }
 
-    const char * find_plug_in_sys_var(const char *name, THD *thd) {
-        const char * result = NULL;
-        String null_string;
+    const char* find_plug_in_sys_var(const char* name, THD* thd) {
+        const char* result = NULL;
         String name_to_find(name, &my_charset_bin);
-        // If get_system_var fails, it calls my_error
-        Item *item = get_system_var(thd,
-                                    OPT_GLOBAL,
-                                    name_to_find.lex_string(),
-                                    null_string.lex_string());
-        if (item) {
-            String scratch;
-            String * str = item->val_str(&scratch);
-            if (str) {
-                result = my_strdup(str->ptr(), MYF(MY_FAE));
-            }
+        LEX_STRING component_name = name_to_find.lex_string();
+
+
+        // 5.7 change the interface to get_system_var and requires a
```

Comment:
why this was get_system_var in the first place? do_deprecated_warning seems redundant? the caller appears to read two string vars and that's all. Why not just find_sys_var, followed by var->value_ptr ?